### ISSUE TITLE
[no-unnecessary-condition] False positive on variable modified in inner scope

### ISSUE THREAD

**Issue by disovi** [typescript-eslint/typescript-eslint#989]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unnecessary-condition": ["error", {"ignoreRhs": true}]
  }
}
```

```TS
let a = false;
const modifyA = (): void => {
  a = true;
};
modifyA();
if (a) { // error
  console.log('a was modified');
}
```

**Expected Result**
no error

**Actual Result**
error  Unnecessary conditional, value is always falsy  @typescript-eslint/no-unnecessary-condition

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.3.0` |
| `@typescript-eslint/parser`        | `2.3.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `6.4.0` |
| `node`                             | `10.16.0` |
| `npm`                              | `6.9.0` |

**Comment by disovi** [comment]
https://github.com/microsoft/TypeScript/issues/9998 original typescript issue
